Add Logic for Geomechanical Use Cases

bska opened the pull request.

bska commented:
jenkins build this please

akva2 commented on opm/elasticity/asmhandler.hpp:
```diff
@@ -110,7 +110,7 @@ class ASMHandler {
                     const LeafIterator& cell,
                     Vector* b=NULL);
-
+    void addMatElement(int i,int j,double val){ A[i][j] += val;}  
```
add spaces

akva2 commented on opm/elasticity/elasticity_impl.hpp:
```diff
@@ -71,6 +71,21 @@ void Elasticity<GridType>::getBmatrix(Dune::FieldMatrix<ctype,components,funcdim
+        = P1ShapeFunctionSet<ctype,ctype,dim>::instance();
+
+    Dune::FieldMatrix<ctype,funcdims,dim> N;
+    for (int i=0;i < funcdims; i++){
```
add space

akva2 commented on opm/elasticity/meshcolorizer.hpp:
```diff
@@ -44,7 +44,14 @@ class MeshColorizer {
 
     //! \brief Calculate the coloring
-    void calcGroups();
+    void calcGroups(){
```
add space

akva2 commented on opm/elasticity/meshcolorizer.hpp:
```diff
@@ -44,7 +44,14 @@ class MeshColorizer {
+    void calcGroups(){
+        tg[0].resize(1);
+        tg[0].reserve(grid.size(0));
+        for (int i = 0; i < grid.size(0); ++i)
```
add braces

akva2 commented on opm/elasticity/meshcolorizer.hpp:
```diff
@@ -44,7 +44,14 @@ class MeshColorizer {
+        for (int i = 0; i < grid.size(0); ++i)
+            tg[0][0].push_back(i);
+        tg[1].clear();
+      
```
remove blank line

akva2 commented on opm/elasticity/meshcolorizer.hpp:
```diff
@@ -44,7 +44,14 @@ class MeshColorizer {
+        tg[1].clear();
+      
+    }
   private:
```
add blank line before private

akva2 reviewed (approved):
Some cosmetics

bska commented:
> Some cosmetics

Thank you for the review.  I've pushed an update that I believe addresses those comments, but please let me know if there are additional steps needed here.

bska commented:
jenkins build this please

akva2 commented:
all good now afaict, so feel free to merge when green.

bska commented:
> feel free to merge when green.

Thanks a lot.  The build check is green now so I'll merge into master.
